[ISSUE 534]: Deprecate conda environment.yml / move to pip

HundredBillion opened the pull request:
This PR resolves [Issue 534](https://github.com/sdatkinson/neural-amp-modeler/issues/534).

Since PyTorch is no longer shipping conda packages, this PR removes the conda environment files and switches the installation docs to a standard Python pip/venv-based setup.

For GPU installs, PyTorch's commands depend on the user's OS and CUDA version, so the docs point to the official PyTorch install page rather than hardcoding a command:

https://pytorch.org/get-started/locally/

For CPU-only and Apple Silicon users, `torch` is already declared in `[project.dependencies]`, so it is installed automatically by `pip install neural-amp-modeler` — no separate PyTorch install step is needed.

### Summary of changes

- Removed `environments/environment_cpu_apple.yml` and `environments/environment_gpu.yml`
- Rewrote `docs/source/installation.rst` to use a 4-step pip/venv install flow with OS-specific venv activation commands (macOS/Linux, Windows cmd.exe, Windows PowerShell)
- Added a `[dev]` optional-dependencies group to `pyproject.toml` (`black`, `flake8`, `pre-commit`) so the full dev environment installs with a single command: `pip install -e ".[dev,test]"`
- Removed the redundant non-GPU `pip install torch` step from the docs, since `torch` is pulled in transitively by `pip install neural-amp-modeler`
- Removed dead conda code from `tests/test_graceful_shutdown.py` (`--conda-env` CLI flag and related branching logic that was never exercised by the pytest path)

HundredBillion committed:
use pip to install pytorch

HundredBillion committed:
update documentation

sdatkinson commented:
Thanks for the PR, @HundredBillion!

This doesn't resolve #534 because installing still uses conda .yml files.

I'll give comments inline presently

sdatkinson commented on docs/source/installation.rst:
```diff
@@ -16,13 +16,11 @@ Step 2: Install NAM
 
 (Windows / Linux users) If your computer has an nVIDIA GPU, you should install a
-GPU-compatible version of PyTorch first:
+GPU-compatible version of PyTorch first. For the recommended installation
```
Don't redirect. Please put the command here.

Usually I'd do like you did, but users of this package have found that too confusing, so for this I prefer to put it more straight in front of them.

sdatkinson commented on environments/environment_cpu_apple.yml:
```diff
@@ -22,9 +21,6 @@ dependencies:
   - pydantic
   - pytest
   - pytest-mock
-  # Performance note:
```
Don't remove this note. It's still relevant.

sdatkinson commented on environments/environment_gpu.yml:
```diff
@@ -37,3 +31,4 @@ dependencies:
       - transformers>=4  # See requirements.txt
       - wavio >=0.0.5
       - -e ..
+      - torch
```
I don't think that this will result in a CUDA-accelerated version of PyTorch being installed?

HundredBillion replied:
ah you're right. I forgot some packages

sdatkinson replied:
Try this:

```
conda env remove -n nam
conda create -f environments/enironment_gpu.yml
conda activate nam
python -c "import torch; assert torch.cuda.is_available()"
```
I expect you'll get an assertion error.

HundredBillion replied:
Ah I understand what you mean. 
Ok I added the command for a CUDA-accelerated version of PyTorch.

HundredBillion committed:
update instruction to install pytorch via pip

HundredBillion commented on docs/source/installation.rst:
```diff
@@ -15,10 +15,18 @@ Step 2: Install NAM
-GPU-compatible version of PyTorch first. For the recommended installation
-commands, please refer to the official PyTorch website:
-https://pytorch.org/get-started/locally/
+.. note::
```
got it.

HundredBillion commented on environments/environment_cpu_apple.yml:
```diff
@@ -33,3 +35,5 @@ dependencies:
       - wavio >=0.0.5
       - -e ..
       - torch
+      - torchvision
```
done

HundredBillion committed:
make PR change request

HundredBillion committed:
create a new file so that CUDA users can still enable CUDA-accelerated PyTorch

HundredBillion committed:
update file to install pytorch with cuda

sdatkinson reviewed (changes requested):
I'm sorry, but this still isn't what I want. There are yml files still in the repo. Moving to pip should see them gone.

HundredBillion commented:
> I'm sorry, but this still isn't what I want. There are yml files still in the repo. Moving to pip should see them gone.

Got it — I misunderstood the issue. I treated it as “use pip inside the conda env files,” but what you want is to remove the conda env approach entirely. I’ll update the PR to delete the environment .yml files and switch the installation docs to a pip/venv-based flow.

HundredBillion committed:
Remove conda environment installs and document pip-based setup

HundredBillion commented on docs/source/installation.rst:
```diff
@@ -3,46 +3,74 @@
 .. code-block:: console
 
-   $ pip install torch --index-url https://download.pytorch.org/whl/cu129
+   $ python -m pip install torch
```
I don’t think we should hardcode a CUDA-specific PyTorch command here since the maintainer asked to defer GPU install instructions to the official PyTorch site

HundredBillion committed:
Clarify PyTorch install docs and use test extra for dev setup

HundredBillion commented on docs/source/installation.rst:
```diff
@@ -3,46 +3,74 @@
+   $ source .venv/bin/activate
+   $ python -m pip install --upgrade pip
+   $ python -m pip install -e .
+   $ python -m pip install pytest pytest-mock flake8 black pre-commit
```
sure - I'll do this.

HundredBillion committed:
Clarify Windows venv activation commands in install docs

HundredBillion commented on docs/source/installation.rst:
```diff
@@ -3,46 +3,74 @@
+   $ python -m pip install --upgrade pip
+
+On Windows:
+
+.. code-block:: console
+
+   > python -m venv .venv
+   > .venv\Scripts\activate
+   > python -m pip install --upgrade pip
+
```
ok - good suggestion

rhaist commented:
Hi @HundredBillion — random passer-by here, also interested in seeing #534 land. A few suggestions if useful, take or leave:

1. **`[dev]` extra in `pyproject.toml`** — I think you mentioned wanting to pick up Copilot's nit about this. Defining the dev tools as an extra keeps the docs from drifting from the project's declared deps, and the local-dev install collapses to one line:

   ```toml
   [project.optional-dependencies]
   dev = [
       "black",
       "flake8",
       "pre-commit",
   ]
   ```

   ```
   $ pip install -e ".[dev,test]"
   ```

2. **The non-GPU `python -m pip install torch` step might be removable.** `torch` is already in `dependencies` in `pyproject.toml`, so `pip install neural-amp-modeler` pulls in the CPU build transitively. That'd make the default-case install genuinely one command, with the PyTorch-site link doing the work for GPU users (which matches what @sdatkinson asked for earlier).

3. **One stray conda reference outside the env files:** `tests/test_graceful_shutdown.py` has a `--conda-env` CLI flag and a `conda_env` kwarg on `run_training_with_interrupt`. The pytest path already hardcodes `None`; only the standalone runner uses it. Might be worth removing in the same PR for consistency, but totally a judgement call.

Tried all three locally on top of `main` (pytest run; same set of unrelated pre-existing failures as baseline, nothing new). Happy to open a follow-up PR if any of this is easier that way.

HundredBillion committed:
Add [dev] optional-dependencies group to pyproject.toml

Moves black, flake8, and pre-commit into a declared [dev] extra so the
dev environment can be set up with a single pip install -e ".[dev,test]".

HundredBillion committed:
Simplify install docs: remove redundant torch step, collapse dev install

Drops the non-GPU "pip install torch" step since torch is already pulled
in transitively by "pip install neural-amp-modeler". Updates the dev
install to a single "pip install -e .[dev,test]" command.

HundredBillion committed:
Remove dead conda code from test_graceful_shutdown.py

Drops the --conda-env CLI flag, conda_env parameter, and conda run
command branch — none of which were used by the pytest path.

HundredBillion commented:
@rhaist - Thanks for your review. I applied all 3 changes you suggested because I think they are all in scope for this PR.
Check out my changes and let me know if it lines up with what you did.
